.buildkite/pipeline.yml

```diff
@@ -697,3 +697,57 @@ steps:
     trigger: "beats-agent-serverless-tests"
     build:
       branch: "main"
+
+  - label: "Trigger DRA (SNAPSHOT)"
+    if: build.pull_request.id != null
+    plugins:
+      - monorepo-diff#v1.0.1:
+          diff: "git diff --name-only origin/${GITHUB_PR_TARGET_BRANCH}...HEAD"
+          interpolation: false
+          watch:
+            - path:
+                - .go-version
+                - .buildkite/packaging.pipeline.yml
+                - .buildkite/pipeline.yml
+                - .buildkite/scripts/dra.sh
+                - .buildkite/scripts/packaging/
+                - dev-tools/packaging/
+              config:
+                trigger: "beats-packaging-pipeline"
+                build:
+                  commit: "${BUILDKITE_COMMIT}"
+                  branch: "${BUILDKITE_BRANCH}"
+                  env:
+                    - BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
+                    - BUILDKITE_PULL_REQUEST_BASE_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
+                    - GITHUB_PR_LABELS=${GITHUB_PR_LABELS}
+                    - DRA_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
+                    - RUN_SNAPSHOT=true
+                    - ELASTIC_SLACK_NOTIFICATIONS_ENABLED=false
+                    - DRY_RUN=true
+
+  - label: "Trigger DRA 7.17 Elastic Agent (SNAPSHOT)"
+    if: (build.pull_request.id != null && build.env('BUILDKITE_PULL_REQUEST_BASE_BRANCH') == "7.17")
+    plugins:
+      - monorepo-diff#v1.0.1:
+          diff: "git diff --name-only origin/${GITHUB_PR_TARGET_BRANCH}...HEAD"
+          interpolation: false
+          watch:
+            - path:
+                - .go-version
+                - .buildkite/pipeline.yml
+                - .buildkite/x-pack/elastic-agent/
+                - dev-tools/packaging/
+              config:
+                trigger: "buildkite-elastic-agent-binary-dra-7-17"
+                build:
+                  commit: "${BUILDKITE_COMMIT}"
+                  branch: "${BUILDKITE_BRANCH}"
+                  env:
+                    - BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
+                    - BUILDKITE_PULL_REQUEST_BASE_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
+                    - GITHUB_PR_LABELS=${GITHUB_PR_LABELS}
+                    - DRA_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
+                    - RUN_SNAPSHOT=true
+                    - ELASTIC_SLACK_NOTIFICATIONS_ENABLED=false
+                    - DRY_RUN=true
```

Comment:
Unfortunately there is a discrepancy in the scripts and this one needs `DRA_DRY_RUN` -> https://github.com/elastic/beats/blob/96fcf35f94c659bba66afb2f16b54a8dbd3987db/.buildkite/x-pack/elastic-agent/scripts/steps/dra-publish.sh#L9